generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs

```diff
@@ -227,17 +227,11 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
       <Col md="8">
         { loading ? <p>Loading...</p> :
         <AvForm model={isNew ? {} : <%= entityInstance %>Entity} onSubmit={saveEntity} >
-          { !isNew ?
-            <AvGroup>
-              <Label for="<%= entityFileName %>-id"><Translate contentKey="global.field.id">ID</Translate></Label>
-              <AvInput id="<%= entityFileName %>-id" type="text" className="form-control" name="id" required readOnly/>
-            </AvGroup>
-            : null
-          }
-          <%_ for (field of fields.filter(field => !field.id)) {
+          <%_ for (field of fields) {
               const fieldType = field.fieldType;
               const fieldName = field.fieldName;
               const fieldNameHumanized = field.fieldNameHumanized;
+              const readonlyIfGeneratedId = (field.id && field.fieldType === 'Long' ? 'readOnly' : '');
```

Comment:
You will probably want to add support to UUID to readOnly too.

Reply:
I think, it's better if I check for databaseType !== sql || field.jpaGeneratedValue - so if the id is not generated by the backend, the user could specify here.